tools/gen_ckan_idx.py

```diff
@@ -1,107 +1,175 @@
-                    "rainfall"
-                ],
-                "notes": f"Radar rainfall dataset for weather event {event}",
-            },
-        }
-        resources = []
-        print(f"event: {event}")
-        print("  netcdf:")
-        for restype in os.listdir(ev_dir):
-            res_dir = os.path.join(ev_dir, restype)
-            if not os.path.isdir(res_dir):
-                continue
-            print("    resource: %r" % (restype,))
-            res_desc = (
-                f"{RESOURCE_DESC_HEAD}. {RESOLUTION_DESC[restype]}. "
-                "Format: NetCDF. Projection: EPSG:3003."
-            )
-            for name in os.listdir(res_dir):
-                url = urljoin(URL_ROOT, os.path.join(
-                    EVENTS_RPATH, event, restype, name)
-                )
-                resources.append({
-                    "url": url,
-                    "name": f"{RESOLUTION_NAME[restype]} Rainfall NetCDF",
-                    "format": "netcdf",
-                    "description": res_desc,
-                })
-        print("  geotiff:")
-        for restype in os.listdir(geo_dir):
-            res_dir = os.path.join(geo_dir, restype)
-            if not os.path.isdir(res_dir):
-                continue
-            print("    resource: %r" % (restype,))
-            res_desc = (
-                f"{RESOURCE_DESC_HEAD}. {RESOLUTION_DESC[restype]}. "
-                "Format: GeoTIFF. Projection: EPSG:4326."
-            )
-            url = urljoin(URL_ROOT, os.path.join(
-                GEOTIFF_RPATH, event, restype, "description.json")
-            )
-            resources.append({
-                "url": url,
-                "name": f"{RESOLUTION_NAME[restype]} Rainfall GeoTIFF",
-                "format": "json",
-                "description": res_desc,
-            })
-        desc["resources"] = resources
-        out_path = os.path.join(OUT_DIR, f"ckan_upload_{event}.json")
-        with io.open(out_path, "wt") as f:
+        rval[date] = entry.path
+    return rval
+
+
+def map_events(stubs, fs_root):
+    m = {date: {"stub": json_path} for date, json_path in stubs.items()}
+    for source, id_list in ID_MAP.items():
+        ds_rpath = DS_RPATH.substitute(source=source)
+        for entry in os.scandir(join(fs_root, ds_rpath)):
+            if entry.name in m:
+                m[entry.name].setdefault("ds", {})[source] = entry.path
+        for id in id_list:
+            ra_rpath = RA_RPATH.substitute(source=source, id=id)
+            for entry in os.scandir(join(fs_root, ra_rpath)):
+                if source == "radar":
+                    date = entry.name
+                else:
+                    name = entry.name.rsplit("_", 1)[-1]
+                    date_obj = strptime(name, "%Y%m%d%H").date()
+                    date = datetime.date.strftime(date_obj, "%Y-%m-%d")
+                ra_map = m[date].setdefault("ra", {})
+                ra_map.setdefault(source, {})[id] = entry.path
```

Comment:
Special case handling sections like this one highlight the current inconsistencies between the different resource trees.